cdap-provisioner-spi/src/main/java/co/cask/cdap/provisioner/spi/ProvisionerSpec.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright © 2018 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.provisioner.spi;
+
+import java.util.Collections;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * Specification of a Provisioner. Includes information about the provisioner as a whole as well as information about
+ * every property supported by the provisioner.
+ */
+public class ProvisionerSpec {
```

Comment:
We use `Specification` in our cdap-api (e.g. `ApplicationSpecification`, `SparkSpecification`, etc). It is better to be consistent.